### PR TITLE
ansible: fix grep error and bool type var

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curve-ansible/roles/format_chunkserver/tasks/include/prepare_chunkserver_with_disk_format.yml [hunk 1]
```diff
@@ -19,10 +19,10 @@
...
-  when: not retain_pool
+  when: not retain_pool | bool
 
 - name: format walfilepool
   shell: sudo ./chunkserver_ctl.sh deploy-wal-pool all --noConfirm
   async: 0
   poll: 3
-  when: not retain_pool and not walfilepool_use_chunk_file_pool
\ No newline at end of file
+  when: not retain_pool | bool and not walfilepool_use_chunk_file_pool | bool
```
Comment (Wine93) [inline curve-ansible/roles/format_chunkserver/tasks/include/prepare_chunkserver_with_disk_format.yml:28]: I think it's better to add some brackets to let the statement more clear.

**Review by Wine93 — COMMENTED** [review]
LGTM

File: curve-ansible/deploy_curve.yml [hunk 2]
```diff
@@ -127,19 +127,19 @@
     - { role: generate_config, template_name: chunkserver.conf, conf_path: "{{ chunkserver_conf_path }}",
                 tags: ["generate_config", "generage_chunkserver_conf"] }
     - { role: generate_config, template_name: client.conf, conf_path: "{{ chunkserver_client_config_path }}",
-                tags: ["generate_config", "generage_chunkserver_conf"] }
+                tags: ["generate_config", "generage_chunkserver_client_conf"] }
```
Comment (cw123) [inline curve-ansible/deploy_curve.yml:130]: Curve-ansible here change some tags. Do you check if any deploy document or wiki need change？

File: curve-ansible/roles/format_chunkserver/tasks/include/prepare_chunkserver_with_disk_format.yml (hunk 1 above)
Comment (SeanHai) [reply curve-ansible/roles/format_chunkserver/tasks/include/prepare_chunkserver_with_disk_format.yml:28]: > I think it's better to add some brackets to let the statement more clear.

done

File: curve-ansible/deploy_curve.yml (hunk 2 above)
Comment (SeanHai) [reply curve-ansible/deploy_curve.yml:130]: > Curve-ansible here change some tags. Do you check if any deploy document or wiki need change？

yes, the doc use like "ansible-playbook deploy_curve.yml -i server.ini --tags generage_chunkserver_conf,generage_chunkserver_client_conf". This issue found in the process of expansion.

**Review by cw123 — APPROVED** [review]

**Review by Wine93 — APPROVED** [review]

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**